### PR TITLE
Add failing test to `matchesKeyInQuery`

### PR DESCRIPTION
**Pull request by JeromeDeLeon**
https://github.com/parse-community/Parse-SDK-JS/issues/911
@dplewis

### CODE REVIEW

**Comment by JeromeDeLeon** [comment]
Same result I got when I ran my own test on it. I even tried to use other events, but no luck.

File: integration/test/ParseLiveQueryTest.js [hunk 1]
```diff
@@ -32,6 +33,35 @@ describe('Parse LiveQuery', () => {
...
+    const subscripton = await query.subscribe();
+
+    subscripton.on('update', object => {
+      assert.equal(object.get('name'), 'tester');
+      done();
+    });
+
+    objectA.set({ name: 'tester' });
+    await objectA.save();
+  });
```
Comment (dplewis) [inline integration/test/ParseLiveQueryTest.js]: Saving object A with tester doesn’t trigger the live query. 

The trigger occurs when the save matches the query you subscribe to. 

I can look more in-depth shortly

Comment (JeromeDeLeon) [reply integration/test/ParseLiveQueryTest.js]: I changed the event to `leave`

**Comment by dplewis** [comment]
Can you check Travis build and lint your code?

**Pull request closed without merging**